addons/lunch/models/lunch_supplier.py

```diff
@@ -123,8 +123,8 @@ def _auto_email_send(self):
 
         for supplier in records:
             send_at = datetime.combine(fields.Date.today(),
-                                       float_to_time(supplier.automatic_email_time, supplier.moment, supplier.tz)).astimezone(pytz.UTC).replace(tzinfo=None)
-            if supplier.available_today and fields.Datetime.now() > send_at:
+                                       float_to_time(supplier.automatic_email_time, supplier.moment, self.env.user.tz))
+            if supplier.available_today and fields.Datetime.now().replace(tzinfo=pytz.timezone(self.env.user.tz)) > send_at:
```

Comment:
I'm not sure this is the correct fix - I believe we should keep the supplier's timezone, and there's another issue with the offset as well.

When a `time()` is converted to a timezone (as done in `float_to_time`), it will take an outdated offset:
```py
>>> time(9, 0).replace(tzinfo=pytz.timezone('Europe/Brussels'))
datetime.time(9, 0, tzinfo=<DstTzInfo 'Europe/Brussels' LMT+0:18:00 STD>)
```

I think it would be better to convert the `send_at` time to UTC like so:
```py
send_time = float_to_time(supplier.automatic_email_time, supplier.moment)
send_at = pytz.timezone(supplier.tz).localize(datetime.combine(fields.Date.today(), send_time)).astimezone(pytz.UTC).replace(tzinfo=None)
```

What do you think? :-)

Reply:
You're right about the offset issue, I messed that up.
However, about the timezone, I find it counter-intuitive. As the user fills the supplier's form, isn't he supposing the order time is in his (user's) timezone?
Maybe @Geoffrey-GMF can help us on that one?

Reply:
The timezone set on the supplier should be used to determine when to send the email - IIRC that field is only shown in debug mode and is set to the user's tz initially.

The issue here is that by using `self.env.user.tz` it will take the timezone of OdooBot as he's the user set on the cron